scripts/kind-with-registry.sh

```diff
@@ -42,37 +56,138 @@ if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true
     registry:2
 fi
 
-# To use workload identity, service account signing key pairs base64 encoded should be exposed via the
-# env variables. The function creates the key pair files after reading it from the env variables.
+# Environment variable inputs
+# SERVICE_ACCOUNT_ISSUER - BYO existing service account issuer
+#    Accepts a URI string, e.g., https://${AZWI_STORAGE_ACCOUNT}.blob.core.windows.net/${AZWI_STORAGE_CONTAINER}/
+#    Assumes that the required openid and jwks artifacts exist in the well-known locations at this URI
+# SERVICE_ACCOUNT_SIGNING_PUB_FILEPATH - A local filepath to a 2048 bit RSA public key
+#    Defaults to capz-wi-sa.pub, must exist locally and match the signed artifacts if using BYO
+# SERVICE_ACCOUNT_SIGNING_KEY_FILEPATH - A local filepath to a 2048 bit RSA private key
+#    Defaults to capz-wi-sa.key, must exist locally and match the signed artifacts if using BYO
+#    If the above keypair filepaths environment variables are not included, a keypair will be created at runtime
+#    Note: if a new keypair is created at runtime then you must not BYO service account issuer
```

Comment:
There is no need to pre-generate the keys, instead you can get the jwks from the cluster after it's created. It's easier and consistent to let kubeadm generate the keys.

```bash
kubectl get --raw /openid/v1/jwks
```

will give you the json that needs to be uploaded to storage account.

Reply:
This would require a change in order:
1. Create storage account or generate the deterministic name.
3. Create kind cluster with the storage account url set as the `--service-account-issuer`.
4. Upload the discovery doc to `/.well-known/openid-configuration`.
5. Get the JWKS from cluster by running `kubectl get --raw /openid/v1/jwks` and upload to `/openid/v1/jwks`.

Reply:
Sorry for the long delay!

So in this case we don't need this config:

```
    extraMounts:
      - hostPath: "${KIND_SERVICE_ACCOUNT_SIGNING_PUB_FILEPATH}"
        containerPath: /etc/kubernetes/pki/sa.pub
      - hostPath: "${KIND_SERVICE_ACCOUNT_SIGNING_KEY_FILEPATH}"
        containerPath: /etc/kubernetes/pki/sa.key
```

Because when we don't include an override to /etc/kubernetes/pki keypair we inherit the auto-generated cluster keypair for free, which has been used to sign the jwks that we're using in the service issuer reference data?

Reply:
that's correct!

Reply:
For a BYO setup, I think the mounts would still be necessary to specify the predefined keys.

Reply:
ACK, thank you. I think I'll advocate we make that change in a follow-up PR just so we're able to merge this ASAP.

Does that sound reasonable?

cc @nojnhuh @mboersma

Reply:
SGTM